clientv3/client_test.go

```diff
@@ -49,7 +49,13 @@ func TestDialCancel(t *testing.T) {
 	c.SetEndpoints("http://254.0.0.1:12345")
 
 	// issue Get to force redial attempts
-	go c.Get(context.TODO(), "abc")
+	getc := make(chan struct{})
+	go func() {
+		defer close(getc)
+		// Get may hang forever on grpc's Stream.Header() if its
+		// context is never canceled.
+		c.Get(c.Ctx(), "abc")
```

Comment:
does `c.Get()` eventually error out when we close client? [`c.Close()`](https://github.com/coreos/etcd/blob/master/clientv3/client.go#L94) -> [`c.conn.Close()`](https://github.com/grpc/grpc-go/blob/777daa17ff9b5daef1cfdf915088a2ada3332bf0/clientconn.go#L565) -> [`balancer.Close()`](https://github.com/coreos/etcd/blob/master/clientv3/balancer.go#L303) 
Then `grpc.Invoke()` from  `kv.Get()` should error out due to [`getTransport()`](https://github.com/grpc/grpc-go/blob/777daa17ff9b5daef1cfdf915088a2ada3332bf0/call.go#L173) error on [`balancer.Get()`](https://github.com/grpc/grpc-go/blob/777daa17ff9b5daef1cfdf915088a2ada3332bf0/clientconn.go#L536)  because balancer is closed.